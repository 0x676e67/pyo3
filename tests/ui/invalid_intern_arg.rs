use pyo3::Python;

fn main() {
    let foo = if true { "foo" } else { "bar" };
    Python::with_gil(|py| py.import(pyo3::intern_bound!(py, foo)).unwrap());
}
