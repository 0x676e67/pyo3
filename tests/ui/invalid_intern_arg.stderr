error[E0435]: attempt to use a non-constant value in a constant
 --> tests/ui/invalid_intern_arg.rs:5:61
  |
5 |     Python::with_gil(|py| py.import(pyo3::intern_bound!(py, foo)).unwrap());
  |                                     ------------------------^^^-
  |                                     |                       |
  |                                     |                       non-constant value
  |                                     help: consider using `let` instead of `static`: `let INTERNED`
